### DISCUSSION TITLE
AssertionError: MultiScaleFlipAug: Must have but only one variable can be set

### DISCUSSION THREAD

**Discussion by SuperFDX** [obss/sahi#804] [Q&A]
@fcakyon Hello, when I use https://github.com/obss/sahi/blob/main/demo/inference_for_mmdetection.ipynb When reasoning the local model, run to result=get_ The prediction ("demo_data/small-vehicles1. jpeg", detection_model) reports an error. The error is as follows:

AssertionError                            Traceback (most recent call last)
File ~/anaconda3/envs/sahi/lib/python3.8/site-packages/mmcv/utils/registry.py:69, in build_from_cfg(cfg, registry, default_args)
     68 try:
---> 69     return obj_cls(**args)
     70 except Exception as e:
     71     # Normal TypeError does not print class name.

File ~/anaconda3/envs/sahi/lib/python3.8/site-packages/mmdet/datasets/pipelines/test_time_aug.py:61, in MultiScaleFlipAug.__init__(self, transforms, img_scale, scale_factor, flip, flip_direction)
     60 self.transforms = Compose(transforms)
---> 61 assert (img_scale is None) ^ (scale_factor is None), (
     62     'Must have but only one variable can be set')
     63 if img_scale is not None:

AssertionError: Must have but only one variable can be set

During handling of the above exception, another exception occurred:

AssertionError                            Traceback (most recent call last)
Cell In[22], line 1
----> 1 result = get_prediction("demo_data/small-vehicles1.jpeg", detection_model)

File ~/anaconda3/envs/sahi/lib/python3.8/site-packages/sahi/predict.py:91, in get_prediction(image, detection_model, shift_amount, full_shape, postprocess, verbose)
     89 # get prediction
     90 time_start = time.time()
...
     70 except Exception as e:
     71     # Normal TypeError does not print class name.
---> 72     raise type(e)(f'{obj_cls.__name__}: {e}')

AssertionError: MultiScaleFlipAug: Must have but only one variable can be set

I searched the Internet and failed to solve the problem. How can I solve it? I look forward to your reply.

**Comment by fcakyon** [comment]
What is your sahi mmdet mmcv torch and cuda versions?

**Reply by SuperFDX** [reply]
Hello, the version of the package I use is as follows:
torch==1.12.1
cuda==11.3
sahi==0.11.9
mmcv-full==1.7.0
mmdet==2.26.0

**Reply by SuperFDX** [reply]
I use https://github.com/obss/sahi/blob/main/demo/inference_for_mmdetection.ipynb
The default model (cascade_mask_rcnn) of the code in the inference source file does not report an error

**Reply by fcakyon** [reply]
Can you link your config? What is your model?

**Reply by SuperFDX** [reply]
My local model is modified based on centripetalNet in mmdetection. I just tested the centripetalNet source code in mmdetection library and reported the same error.
https://github.com/open-mmlab/mmdetection/blob/master/configs/centripetalnet/centripetalnet_hourglass104_mstest_16x6_210e_coco.py

**Reply by SuperFDX** [reply]
I think it is caused by the parameters in the train pipeline in its configuration file, but I don't know how to solve it.

**Reply by SuperFDX** [reply]
@fcakyon  Hello, how can I solve this problem? We look forward to your reply

**Reply by fcakyon** [reply]
This seems like a mmdetection issue. You should open an issue on that repo.

**Reply by SuperFDX** [reply]
However, the source code can be trained normally, and only when using sahi for reasoning can an error be reported.

**Comment by z-hwa** [comment]
For mmdetection 2.
You can find models\mmdet.py in sahi 0.11.14, which is below code
```
def load_model(self):
        from mmdet.apis import init_detector

        # create model
        model = init_detector(
            config=self.config_path,
            checkpoint=self.model_path,
            device=self.device,
        )

        # update model image size
        if self.image_size is not None:
            # print('\n\n\n' + str(model.cfg.data.test.pipeline[1]) + '\n\n\n\n\n')
            model.cfg.data.test.pipeline[1]["img_scale"] = (self.image_size, self.image_size)

        self.set_model(model)
```

The key point is here, which will modify your config's multiscaling.
So ensure you do not use scale_factor, and using any scale with img_scale.
Check the setting both in your config's data part and pipeline part.
`model.cfg.data.test.pipeline[1]["img_scale"] = (self.image_size, self.image_size)`
